common/db/job_store/dynamo.go

```diff
@@ -73,9 +69,34 @@ func (d *DynamoJobStore) UpdateJobStatus(jobID string, status types.JobStatus) e
 }
 
 func (d *DynamoJobStore) SetJobMeta(jobID string, meta map[string]string) error {
+	// ensure we perform a consistent read after a write
+	defer d.setConsistentRead(true)
+
 	if err := d.table.Update("JobID", jobID).Set("Meta", meta).Run(); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (d *DynamoJobStore) SelectAll() ([]*models.Job, error) {
+	defer d.setConsistentRead(false)
+
+	jobs := []*models.Job{}
+	if err := d.table.Scan().Consistent(d.consistentRead).All(&jobs); err != nil {
+		return nil, err
+	}
+
+	return jobs, nil
+}
+
+func (d *DynamoJobStore) SelectByID(jobID string) (*models.Job, error) {
+	defer d.setConsistentRead(false)
+
+	var job *models.Job
+	if err := d.table.Get("JobID", jobID).Consistent(d.consistentRead).One(&job); err != nil {
+		return nil, err
```

Comment:
Since `JobID` is a unique identifier on this table the result should either be one item or no items. DynamoDB is very good at fetching a single item on an indexed field. 

To simplify your application design I would always use a consistent read when fetching a single item by its unique identifier.